### ISSUE TITLE
EOF Error thrown by Rack

### ISSUE THREAD

**Issue by fosterfarrell9** [MaMpf-HD/mampf#175]
We are occasionally seing exceptions in production with stack trace like this.

```
 EOFError
/usr/local/bundle/gems/rack-2.2.3/lib/rack/multipart/parser.rb:359:in `handle_empty_content!'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/multipart/parser.rb:186:in `on_read'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/multipart/parser.rb:77:in `block in parse'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/multipart/parser.rb:75:in `loop'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/multipart/parser.rb:75:in `parse'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/multipart.rb:54:in `extract_multipart'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/request.rb:594:in `parse_multipart'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/request.rb:446:in `POST'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/request.rb:469:in `params'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/request.rb:32:in `params'
/usr/local/bundle/gems/shrine-3.3.0/lib/shrine/plugins/upload_endpoint.rb:130:in `get_multipart_upload'
/usr/local/bundle/gems/shrine-3.3.0/lib/shrine/plugins/upload_endpoint.rb:119:in `get_io'
/usr/local/bundle/gems/shrine-3.3.0/lib/shrine/plugins/upload_endpoint.rb:109:in `handle_request'
/usr/local/bundle/gems/shrine-3.3.0/lib/shrine/plugins/upload_endpoint.rb:91:in `block in call'
/usr/local/bundle/gems/shrine-3.3.0/lib/shrine/plugins/upload_endpoint.rb:87:in `catch'
/usr/local/bundle/gems/shrine-3.3.0/lib/shrine/plugins/upload_endpoint.rb:87:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/routing/mapper.rb:19:in `block in <class:Constraints>'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/routing/mapper.rb:48:in `serve'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/journey/router.rb:49:in `block in serve'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/journey/router.rb:32:in `each'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/journey/router.rb:32:in `serve'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/routing/route_set.rb:834:in `call'
/usr/local/bundle/gems/warden-1.2.9/lib/warden/manager.rb:36:in `block in call'
/usr/local/bundle/gems/warden-1.2.9/lib/warden/manager.rb:34:in `catch'
/usr/local/bundle/gems/warden-1.2.9/lib/warden/manager.rb:34:in `call'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/tempfile_reaper.rb:15:in `call'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/etag.rb:27:in `call'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/conditional_get.rb:40:in `call'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/head.rb:12:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/http/content_security_policy.rb:18:in `call'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/session/abstract/id.rb:266:in `context'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/session/abstract/id.rb:260:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/middleware/cookies.rb:648:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
/usr/local/bundle/gems/activesupport-6.0.3.4/lib/active_support/callbacks.rb:101:in `run_callbacks'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/middleware/callbacks.rb:26:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/middleware/actionable_exceptions.rb:18:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/middleware/debug_exceptions.rb:32:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
/usr/local/bundle/gems/railties-6.0.3.4/lib/rails/rack/logger.rb:37:in `call_app'
/usr/local/bundle/gems/railties-6.0.3.4/lib/rails/rack/logger.rb:26:in `block in call'
/usr/local/bundle/gems/activesupport-6.0.3.4/lib/active_support/tagged_logging.rb:80:in `block in tagged'
/usr/local/bundle/gems/activesupport-6.0.3.4/lib/active_support/tagged_logging.rb:28:in `tagged'
/usr/local/bundle/gems/activesupport-6.0.3.4/lib/active_support/tagged_logging.rb:80:in `tagged'
/usr/local/bundle/gems/railties-6.0.3.4/lib/rails/rack/logger.rb:26:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
/usr/local/bundle/gems/request_store-1.5.0/lib/request_store/middleware.rb:19:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/middleware/request_id.rb:27:in `call'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/method_override.rb:24:in `call'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/runtime.rb:22:in `call'
/usr/local/bundle/gems/activesupport-6.0.3.4/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/middleware/executor.rb:14:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/middleware/static.rb:126:in `call'
/usr/local/bundle/gems/rack-2.2.3/lib/rack/sendfile.rb:110:in `call'
/usr/local/bundle/gems/actionpack-6.0.3.4/lib/action_dispatch/middleware/host_authorization.rb:76:in `call'
/usr/local/bundle/gems/railties-6.0.3.4/lib/rails/engine.rb:527:in `call'
/usr/local/bundle/gems/puma-4.3.6/lib/puma/configuration.rb:228:in `call'
/usr/local/bundle/gems/puma-4.3.6/lib/puma/server.rb:713:in `handle_request'
/usr/local/bundle/gems/puma-4.3.6/lib/puma/server.rb:472:in `process_client'
/usr/local/bundle/gems/puma-4.3.6/lib/puma/server.rb:328:in `block in run'
/usr/local/bundle/gems/puma-4.3.6/lib/puma/thread_pool.rb:134:in `block in spawn_thread'
```

It might be related to [this issue](https://github.com/rack/rack/issues/903) in Rack (where several suggestions are made on how to work around).

**Comment by Splines** [comment]
Is this still an issue?

**Comment by fosterfarrell9** [comment]
No, this was fixed in #552.
